frontend/src/pages/notebookController/screens/server/SizeSelectField.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { FormGroup, Select, SelectOption } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import AppContext from '../../../../app/AppContext';
+
+type SizeSelectFieldProps = {
+  value: string;
+  setValue: (newValue: string) => void;
+};
+
+const SizeSelectField: React.FC<SizeSelectFieldProps> = ({ value, setValue }) => {
+  const [sizeDropdownOpen, setSizeDropdownOpen] = React.useState<boolean>(false);
+  const { dashboardConfig } = React.useContext(AppContext);
+  const sizes = dashboardConfig?.spec?.notebookSizes || [];
+
+  const sizeOptions = () => {
+    if (sizes.length === 0) {
+      return [<SelectOption key="Default" value="No sizes" />];
+    }
+
+    return sizes.map((size) => {
+      const name = size.name;
+      const desc =
+        `Limits: ${size?.resources?.limits?.cpu || '??'} CPU, ` +
+        `${size?.resources?.limits?.memory || '??'} Memory ` +
+        `Requests: ${size?.resources?.requests?.cpu || '??'} CPU, ` +
+        `${size?.resources?.requests?.memory || '??'} Memory`;
+      return <SelectOption key={name} value={name} description={desc} />;
```

Comment:
Interesting -- we should probably make our CRD force `requests` and `limits` when they add the items. I logged https://github.com/opendatahub-io/odh-dashboard/issues/420 to manage it.